libsolidity/parsing/Parser.h

```diff
@@ -102,7 +102,8 @@ class Parser: public langutil::ParserBase
 	ASTPointer<OverrideSpecifier> parseOverrideSpecifier();
 	StateMutability parseStateMutability();
 	FunctionHeaderParserResult parseFunctionHeader(bool _isStateVariable);
-	ASTPointer<ASTNode> parseFunctionDefinition(bool _freeFunction = false, bool _allowBody = true);
+	ASTPointer<ForAllQuantifier> parseQuantifiedFunctionDefinition();
+	ASTPointer<FunctionDefinition> parseFunctionDefinition(bool _freeFunction = false, bool _allowBody = true);
```

Comment:
This is quite a sneaky change in here :-). Yes, it does have its own commit... but it still doesn't really belong in this PR and such :-)... but ok, I won't complain about it this time, let's just keep it :-).

Reply:
It's related. Without this I'd have to cast the result of `parseFunctionDefinition()` to `FunctionDefinition` in the code I added. I could do that, but come on, the function is called `parseFunctionDefinition()` and it does not make sense for it to return anything other than a `FunctionDefinition`.

Reply:
Ok, fair enough. Not that in the long run the forall node shouldn't probably take a more general type than a function definition, but fine to keep both as is now.